src/pytrajplot/plotting/analyse_trajectories.py

```diff
@@ -108,7 +109,11 @@ def _check_dateline_crossing(
         # 2) check if the any of the sign_flips are relevant (i.e. not crossing the 0° longitude line)
         for flip_index in sign_flip_indexes:
             # print(f"lon before/after flip: (index: {flip_index})\t{lon[flip_index-1]} --> {lon[flip_index]}")
-            if not np.isnan(lon[flip_index]) and not np.isnan(lon[flip_index - 1]):
+            if (
+                not np.isnan(lon[flip_index])
+                and not np.isnan(lon[flip_index - 1])
+                and flip_index % number_of_times != 0
```

Comment:
I guess this is the core of the bug fix, and I am lacking knowhow to properly review. 
I find `number_of_times` a confusing name that does not help understand the logic. From a different part of the code I read `#lines that make up one trajectory`. So would it be more appropriate to say `number_of_segments` ? 
Is a trajectory composed by a number of segments with all same number of points and the condition does want to ignore when moving between segments? 
If so, I would probably solve the problem differently by using a different data structure, a sequence of sequences for lon. So that the algorithms applied to only the inner most sequence can not mixed accross contiguous segments. That eliminates the need of this extra condition, and makes the intention more readable. 
But I am not sure I understood it

Reply:
number_of_times is what is really meant, as a trajectory is composed of a (fixed per file) number of time steps, however restructuring the data as proposed could be beneficial --> created an issue for later